packages/backend/src/models/SchedulerModel/index.ts

```diff
@@ -444,6 +444,16 @@ export class SchedulerModel {
     }
 
     async logSchedulerJob(log: SchedulerLog): Promise<void> {
+        if (log.schedulerUuid) {
```

Comment:
If I'm correct, most of the time this check will pass? In that case we should skip this asking for permission and just try the insert.

If the insert throws, we can check the error code, (e.g. `23503` is a foreign key violation). If it's because of the missing foreign key we could ignore, otherwise rethrow. 

As it is, there is still a race condition where I could delete the scheduler between the check and the insert (as well as the overhead of doing an extra db call)

Reply:
> If I'm correct, most of the time this check will pass? In that case we should skip this asking for permission and just try the insert.

yeah, that's correct, schedulerUuid will be present in SchedulerLog, but it fails on DB insert.

I am happy with your error code suggestion 

Question, if this happens on let's say `STARTED` does it make sense to even continue the job ?

Reply:
thanks both for the comments, I have rewritten the code to check for the specific constraint!

Reply:
and changed back to the error code check!